lib/livebook_web/live/sidebar_helpers.ex

```diff
@@ -121,13 +121,21 @@ defmodule LivebookWeb.SidebarHelpers do
   end
 
   def shared_home_handlers(socket) do
-    attach_hook(socket, :shutdown, :handle_event, fn
-      "shutdown", _params, socket ->
-        System.stop()
-        {:halt, put_flash(socket, :info, "Livebook is shutting down. You can close this page.")}
-
-      _event, _params, socket ->
-        {:cont, socket}
-    end)
+    if shutdown_enabled?() do
+      attach_hook(socket, :shutdown, :handle_event, fn
+        "shutdown", _params, socket ->
+          System.stop()
+          {:halt, put_flash(socket, :info, "Livebook is shutting down. You can close this page.")}
+
+        _event, _params, socket ->
+          {:cont, socket}
+      end)
+    else
+      socket
+    end
+  end
+
+  defp shutdown_enabled? do
+    Application.fetch_env!(:livebook, :shutdown_enabled)
   end
```

Comment:
We generally read from the env in `Livebook.Config`, so maybe move there?

Reply:
Sounds good. Can you please do this change and merge? :)

Reply:
Or merge and then change it!

Reply:
Surething!